File: api/editor/api-editor/src/main/java/ca/bc/gov/tno/areas/editor/models/FileReferenceModel.java
package ca.bc.gov.tno.areas.editor.models;

import ca.bc.gov.tno.dal.db.entities.FileReference;
import ca.bc.gov.tno.models.AuditColumnModel;

public class FileReferenceModel extends AuditColumnModel {
  /**
   * Primary key to identify the file reference.
   */
  private int id;

  /**
   * The file mime-type.
   */
  private String mimeType;

  /**
   * The path to the file.
   */
  private String path;

  /**
   * The size of the file in bytes.
   */
  private int size;

  /**
   * The length of the video/audio in milliseconds.
   */
  private int runningTime;

  public FileReferenceModel() {
  }

  public FileReferenceModel(FileReference entity) {
    super(entity);

    if (entity != null) {
      this.id = entity.getId();
      this.mimeType = entity.getMimeType();
      this.path = entity.getPath();
      this.size = entity.getSize();
      this.runningTime = entity.getRunningTime();
    }
  }

  /**
   * @return int return the id
   */
  public int getId() {
    return id;
  }

  /**
   * @param id the id to set
   */
  public void setId(int id) {
    this.id = id;
  }

  /**
   * @return String return the mimeType
   */
  public String getMimeType() {
    return mimeType;
  }

  /**
   * @param mimeType the mimeType to set
   */
  public void setMimeType(String mimeType) {
    this.mimeType = mimeType;
  }

  /**
   * @return String return the path
   */
  public String getPath() {
    return path;
  }

  /**
   * @param path the path to set
   */
  public void setPath(String path) {
    this.path = path;
  }

  /**
   * @return int return the size
   */
  public int getSize() {
    return size;
  }

  /**
   * @param size the size to set
   */
  public void setSize(int size) {
    this.size = size;
  }

  /**
   * @return int return the runningTime
   */
  public int getRunningTime() {
    return runningTime;
  }

  /**
   * @param runningTime the runningTime to set
   */
  public void setRunningTime(int runningTime) {
    this.runningTime = runningTime;
  }

}

File: api/net/Areas/Editor/Models/Content/FileReferenceModel.cs
using TNO.API.Models;

namespace TNO.API.Areas.Editor.Models.Content;

/// <summary>
/// FileReferenceModel class, provides a model that represents an file reference.
/// </summary>
public class FileReferenceModel : AuditColumnsModel
{
    #region Properties
    /// <summary>
    /// get/set - Primary key to file reference.
    /// </summary>
    public long Id { get; set; }

    /// <summary>
    /// get/set - Foreign key to content.
    /// </summary>
    public long ContentId { get; set; }

    /// <summary>
    /// get/set - The file mime type.
    /// </summary>
    public string MimeType { get; set; } = "";

    /// <summary>
    /// get/set - The path to the file.
    /// </summary>
    public string Path { get; set; } = "";

    /// <summary>
    /// get/set - The size of the file in bytes.
    /// </summary>
    public int Size { get; set; }

    /// <summary>
    /// get/set - The number of seconds this file runs.
    /// </summary>
    public int RunningTime { get; set; }
    #endregion

    #region Constructors
    /// <summary>
    /// Creates a new instance of an FileReferenceModel.
    /// </summary>
    public FileReferenceModel() { }

    /// <summary>
    /// Creates a new instance of an FileReferenceModel, initializes with specified parameter.
    /// </summary>
    /// <param name="entity"></param>
    public FileReferenceModel(Entities.FileReference entity) : base(entity)
    {
        this.Id = entity.Id;
        this.ContentId = entity.ContentId;
        this.MimeType = entity.MimeType;
        this.Path = entity.Path;
        this.Size = entity.Size;
        this.RunningTime = entity.RunningTime;
    }
    #endregion

    #region Methods
    /// <summary>
    /// Creates a new instance of a FileReference object.
    /// </summary>
    /// <param name="contentId"></param>
    /// <returns></returns>
    public Entities.FileReference ToEntity(long contentId)
    {
        var entity = (Entities.FileReference)this;
        entity.ContentId = contentId;
        return entity;
    }

    /// <summary>
    /// Explicit cast to entity.
    /// </summary>
    /// <param name="model"></param>
    public static explicit operator Entities.FileReference(FileReferenceModel model)
    {
        return new Entities.FileReference(model.ContentId, model.MimeType, model.Path)
        {
            Id = model.Id,
            Size = model.Size,
            RunningTime = model.RunningTime,
            Version = model.Version ?? 0,
        };
    }
    #endregion
}
